src/modals/examples/modal-overlay.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * @ngdoc directive
+ * @name patternfly.modals.componenet:pfModalOverlay
+ * @element pfModalOverlay
+ * @restrict E
+ *
+ * @description
+ * The Modal Overlay pattern provides a way to quickly load and display important information to the user without navigating
+ * away from the current page. It utilizes the {@link https://angular-ui.github.io/bootstrap/#modal Angular UI modal}
+ * and Patternfly styling. Use cases for Modal Overlays can vary a great deal, but some examples include the following:
+ * <ul>
+ * <li>To remind or prompt users
+ * <li>To load dialogs that require user input before advancing
+ * <li>To load tasks which require a user’s full attention, such as stepping through a wizard flow
+ * <li>To present important information or warnings
+ * </ul>
+ *
+ * @param {boolean} showModal Flag for setting modal to hide/show by default
+ * @param {function} onClose Function to call when the modal is closed<br/>
+ * Note: the parameter passed to the on-close function must be named 'dismissCause'
```

Comment:
why?

Reply:
because this is the named pair we used in your comment below.  Has to do with the binding being `close: "&onClose",`